osquery/tables/system/linux/md_tables.cpp

```diff
@@ -0,0 +1,735 @@
+          // Add an extra line for next iteration if so..
+          n += 1;
+
+        } else if ((pos = lines[n + 1].find("check =")) != std::string::npos) {
+          std::string checkArray(
+              lines[n + 1].substr(pos + sizeof("check =") - 1));
+          boost::algorithm::trim(checkArray);
+          parseMDAction(checkArray, mdd.checkArray);
+          // Add an extra line for next iteration if so..
+          n += 1;
+
+        } else if ((pos = lines[n + 1].find("bitmap:")) != std::string::npos) {
+          std::string bitmap(lines[n + 1].substr(pos + sizeof("bitmap:") - 1));
+          boost::algorithm::trim(bitmap);
+          parseMDBitmap(bitmap, mdd.bitmap);
+          // Add an extra line for next iteration if so..
+          n += 1;
+          // If none of above, then we can break out of loop
+        } else {
+          break;
+        }
+      }
+
+      result.devices.push_back(mdd);
+
+      // Assume unused
+    } else if (firstTwo == "un") {
+      result.unused = lines[n].substr(sizeof("unused devices:") - 1);
+      boost::algorithm::trim(result.unused);
+      // Unexpected mdstat line, log a warning...
+    } else {
+      LOG(WARNING) << "Unexpected mdstat line: " << lines[n];
+    }
+
+    n += 1;
+  }
+}
+
+void getDrivesForArray(std::string arrayName,
+                       MDInterface& md,
+                       QueryData& data) {
+  std::string path(md.getPathByDevName(arrayName));
+  if (path == "") {
+    LOG(ERROR) << "Could not get file path for " << arrayName;
+    return;
+  }
+
+  mdu_array_info_t array;
+  if (md.getArrayInfo(path, array) != true) {
+    return;
+  }
+
+  QueryData temp;
+  for (size_t i = 0; i < MD_SB_DISKS; i++) {
+    mdu_disk_info_t disk;
+    disk.number = i;
+    if (md.getDiskInfo(path, disk) != true) {
+      continue;
+    }
+
+    if (disk.major > 0) {
+      Row r;
+      r["md_device_name"] = arrayName;
+      r["drive_name"] = md.getDevName(disk.major, disk.minor);
+      r["state"] = getDiskStateStr(disk.state);
+      r["slot"] = INTEGER(disk.raid_disk);
+
+      /* We have to check here b/c otherwise we have no idea if the slot has
+       * been recovered.  We assume that if the disk number is less than the
+       * total disk count of the array, that the original slot position;  If the
+       * number is greater than the disk count, then it's not safe to make that
+       * assumption, and we lose precision on the missing slot resolution in the
+       * below block. */
+      if (disk.raid_disk < 0 && disk.number < array.raid_disks) {
+        r["slot"] = std::to_string(disk.number);
+      }
+
+      temp.push_back(r);
+    }
+  }
+
+  // Find removed disks if number of rows don't match with array raid disks
+  for (int slot = 0; slot < array.raid_disks; slot++) {
+    bool found = false;
+    int softRemoved = -1;
+
+    for (size_t i = 0; i < temp.size(); i++) {
+      if (std::stoi(temp[i]["slot"]) == slot) {
+        found = true;
```

Comment:
I'm not 100% on what you're searching for here, but instead of walking `raid_disks` then the `temp`, can you keep a `map` of the value you need to lookup?